### DISCUSSION TITLE
opto avec cobrapin

### DISCUSSION THREAD

**Discussion by debiasfloriane** [missionpinball/missionpinball#63] [Q&A]
Hello,
I'm starting to build a pinball machine and I have a problem. I have a cobrapin card. I have an opto switch for a ramp like this: https://www.pinballlife.com/data-eastsegastern-opto-transceiver.html. I also have a ball pit assembly like this: https://www.pinballlife.com/ball-trough-assembly-4-ball.html.
None of the optical sensors are working.
For that of the ramp, the infrared LED lights up very strongly in red. for that of the launcher I see nothing (nor with my smartphone).
in my config file they are connected to coils which suddenly fire every 15 sec (there is no change of active switches, inactive switches in my log). They are connected to direct inputs. the transmitters are plugged into a 5v supply with a 100ohm resistor. The receivers are on the GND of J1 and the direct inputs of J1. the cobrapin board is 48v powered for the coils but not 5v powered.
Is there a way to test optical switches? with a multimeter? with mfp?
Thanks in advance

**Comment by cobra18t** [comment]
Starting with the trough, You should be able to see something at least with a cell phone. You have it connected similar to this? https://groups.google.com/g/mpf-users/c/Cn_2ZoLyBpY/m/xMRtTYOJAQAJ

The recommended current for the trough transmitters is around 75mA, but your 100ohm only gives 33mA. I recommend another 100ohm in parallel for a resulting 66mA. Triple check your pinout and share a rough diagram of your connections if possible.

On to the ramp transceiver...

I am not sure that ramp transceiver will work directly with CobraPin. In looking at the manuals of Stern machines that use that transceiver (see page 136 of the WPT manual), there is an additional amplifier board (520-5239-01) used to properly condition the output of that receiver. 

They are using a photodiode instead of a phototransistor. The photodiode is cool because you can use the same part to send or receive, but it comes at the cost of more circuitry to condition the receive side.

**Reply by cobra18t** [reply]
A Williams pair that uses a phototransistor for the receiver may be a better option for the ramp opto: https://www.pinballlife.com/williamsbally-infared-led-opto-assembly-set.html

You still need to current limit the transmitter with a resistor.

**Comment by debiasfloriane** [comment]
yes I plugged it like that. I will change the IR diodes on the transmitter side. I tried with 2 resistances of 100 ohm or 1 of 220 Ohm but nothing. for the ramp I'll take that. it will be easier.
https://pinballmedics.ca/product/complete-spinner-assembly-with-switch-left-side/
thank you for the answer
